### PR TITLE
Patterns: Add Patterns to universal header and footer navigation

### PR DESCRIPTION
**Pull request by stephanethomas**
This pull request addresses https://github.com/Automattic/dotcom-forge/issues/6373 by adding a `WordPress Patterns` item to the `Features` section in the header and footer navigation (just after `WordPress Plugins`):

| Header | Footer
| -----  | -----
| image | image

It also fixes inconsistencies and strings that were not correctly localized.

#### Testing instructions

1. Run `git checkout add/patterns-to-header-and-footer` and start your server, or access a [live branch](https://github.com/Automattic/wp-calypso/pull/89122#issuecomment-2032517498)
2. Make sure you are logged out
3. Open the [`Themes` page](http://calypso.localhost:3000/themes)
4. Assert that there is a `WordPress Patterns` link in the `Features` menu in the header
5. Assert that there is a `WordPress Patterns` link in the `Features` section in the footer
6. Log into WordPress.com now
7. Open the [`Patterns` page](http://calypso.localhost:3000/patterns)
8. Repeat steps `#4` and `#5`

### CODE REVIEW

File: apps/happy-blocks/block-library/universal-header/index.php [hunk 1]
```diff
@@ -498,6 +505,13 @@ class="x-icon x-icon--external" role="img" aria-hidden="true" focusable="false">
 									<?php echo esc_html( fixme__( 'WordPress Plugins', __( 'Plugins', 'happy-blocks' ) ) ); ?>
 								</a>
 							</li>
+							<li class="x-menu-grid-item">
+								<a role="menuitem" class="x-menu-link x-link"
+								   href="<?php echo esc_url( localized_wpcom_url( 'https://wordpress.com/patterns' ) ); ?>"
+								   tabindex="-1">
+									<?php echo esc_html( fixme__( 'WordPress Patterns', __( 'Patterns', 'happy-blocks' ) ) ); ?>
+								</a>
+							</li>
```
Comment (stephanethomas) [inline apps/happy-blocks/block-library/universal-header/index.php:514]: My understanding is that those components are used on https://wordpress.com/support but I wasn't able to find comprehensive testing instructions. However, given how simple those changes are I think visual inspection is enough.

Comment (chad1008) [reply apps/happy-blocks/block-library/universal-header/index.php:514]: Also not familiar with them, but as you say this looks simple enough. It also matches the existing links perfectly so I'd agree we're good!

**Review by chad1008 — APPROVED** [review]
LGTM :rocket: :ship:

File: packages/wpcom-template-parts/src/universal-footer-navigation/index.tsx [hunk 2]
```diff
@@ -121,14 +121,6 @@ export const PureUniversalNavbarFooter = ( {
...
 									</a>
 								</li>
-								<li>
-									<a href="https://wpvip.com/" data-is_external="1" target="_self">
-										{ __( 'Enterprise', __i18n_text_domain__ ) }{ ' ' }
-										<span className="lp-link-chevron-external">
-											{ __( 'Solutions', __i18n_text_domain__ ) }
-										</span>
-									</a>
-								</li>
```
Comment (stephanethomas) [inline packages/wpcom-template-parts/src/universal-footer-navigation/index.tsx:131]: I've renamed `Enterprise Solutions` to `Enterprise` to match the label used in the header. I've also move it below `Website Design Services` to be consistent with footers from landing pages (such as https://wordpress.com/support/).

**Pull request merged**